internal/capability/operator_install_test.go

```diff
@@ -0,0 +1,90 @@
+package capability
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/opdev/opcap/internal/operator"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Operator install tests", func() {
+	os.Chdir(os.TempDir())
+
+	csv := operatorv1alpha1.ClusterServiceVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testcsv",
+			Namespace: "testns",
+		},
+		Status: operatorv1alpha1.ClusterServiceVersionStatus{
+			Phase: operatorv1alpha1.CSVPhaseSucceeded,
+		},
+	}
+
+	fakeClient := operator.NewFakeOpClient(&csv)
+
+	operatorGroupData := operator.OperatorGroupData{
+		Name:             "testog",
+		TargetNamespaces: []string{"default", "testns"},
+	}
+
+	subscription := operator.SubscriptionData{
+		Name:            "testsub",
+		Channel:         "test",
+		InstallModeType: v1alpha1.InstallModeTypeAllNamespaces,
+		Package:         "testpackage",
+		CatalogSource:   "testcatalog",
+	}
+
+	customResources := []map[string]interface{}{
+		{
+			"kind": "testkind",
+			"metadata": map[string]interface{}{
+				"name": "testname",
+			},
+		},
+	}
+
+	auditFn, auditCleanupFn := operatorInstall(context.TODO(),
+		withClient(fakeClient),
+		withNamespace("testingThings"),
+		withOperatorGroupData(&operatorGroupData),
+		withSubscription(&subscription),
+		withTimeout(1),
+		// might need this
+		withCustomResources(customResources),
+	)
+
+	Context("creating a new operator install audit", func() {
+		When("given valid options", func() {
+			It("should return functional auditFn and auditCleanupFn", func() {
+				// add /tmp
+
+				Expect(auditFn(context.TODO())).To(Succeed())
+				// test contents of fake Client actually changed/were added
+
```

Comment:
I assume you are planning to add more tests? Are you planning to do that in this PR, or a followup?

Reply:
same PR, just wanted to get some initial feedback